framework/global/val.h

```diff
@@ -47,29 +48,31 @@ class Val
     explicit Val(double val);
     explicit Val(bool val);
     explicit Val(int val);
-    explicit Val(QColor val);
 
     void setType(Type t);
     Type type() const;
 
     bool isNull() const;
-    const std::string& toString() const;
+    std::string toString() const;
     double toDouble() const;
     bool toBool() const;
     int toInt() const;
 
 #ifndef NO_QT_SUPPORT
+    explicit Val(QColor color);
+    explicit Val(QVariant val);
+
     QColor toQColor() const;
     QString toQString() const;
     QVariant toQVariant() const;
+
     static Val fromQVariant(const QVariant& var);
 #endif
 
     inline bool operator ==(const Val& v) const { return v.m_val == m_val && v.m_type == m_type; }
 
 private:
-
-    std::string m_val;    //! NOTE In C++17 can be replaced by std::any or std::variant
+    QVariant m_val; //! NOTE In C++17 can be replaced by std::any or std::variant
```

Comment:
Not a good idea.
This means that there is a strong dependence on Qt.
The very meaning of the `Val` class is that there is no dependence on Qt. Otherwise, instead of `Val`, one could simply use `QVariant`

Reply:
I discussed this issue with Igor in Slack. We decided to leave my current solution as is because in some cases Val must storage some complex types (like QVariantMap) and currently there is no better ability to do this